src/value.c

```diff
@@ -372,16 +358,14 @@ int SIValue_ToDouble(SIValue *v, double *d) {
   }
 }
 
-void SIValue_FromString(SIValue *v, char *s, size_t s_len) {
+void SIValue_FromString(SIValue *v, char *s) {
   int numeric = 1;
-  int i;
-  char c;
+  char *c;
 
-  /* Scan string, see if we can find any none numeric characters int it. */
-  for(i = 0; i < s_len; i++) {
-    c = s[i];
+  /* Scan string, see if we can find any non-numeric characters in it. */
+  for(c = s; *c != '\0'; c ++) {
     
-    if(!isdigit(c) && c != '.' && c != '-' && c != '+') {
+    if(!isdigit(*c) && *c != '.' && *c != '-') {
```

Comment:
You're right, that was unintentional!

I think we should actually replace this entire loop with a `strtod` call, as it will handle all of these cases as well as values with exponents (`+1.5e5`, for example, is a valid conversion target).

As this function only constructs string and double types, we can then actually drop the `SI_ParseValue` call entirely.

Reply:
Agree,
As I see it, we need to determine if s represents a number or a string,
if s is numeric return SI_DoubleVal otherwise return SI_StringVal

Change signature to:
SIValue SIValue_FromString(char *s)

What do you think?